src/scene/Scene.ts

```diff
@@ -91,12 +98,29 @@ class Scene {
   }
 
   replaceAllElements(nextElements: readonly ExcalidrawElement[]) {
-    this.elements = nextElements;
+    this.elements = [];
+    const elements: ExcalidrawElement[] = [];
     this.elementsMap.clear();
+    const elementsToBeStackedOnTop: ExcalidrawElement[] = [];
     nextElements.forEach((element) => {
+      if (isCustomElement(element)) {
+        const config = getCustomElementConfig(
+          this.app.props.customElementsConfig,
+          element.customType,
+        );
+        if (config?.stackedOnTop) {
+          elementsToBeStackedOnTop.push(element);
+        } else {
+          elements.push(element);
+        }
+      } else {
+        elements.push(element);
+      }
       this.elementsMap.set(element.id, element);
       Scene.mapElementToScene(element, this);
     });
+    elementsToBeStackedOnTop.forEach((ele) => elements.push(ele));
```

Comment:
I don't like this, though I'm not sure what other quick hack we can do instead.

But, we can't be looping through potentially large configs for each custom element (107). Also let's just concat the two arrays instead of another loop (122).

Still, we should figure out something better than this (keep in mind that in collab we call this on every update. This api should be kept fast).